tests/conftest.py

```diff
@@ -124,14 +124,26 @@ def subfolder_run_directory(tmpdir: pathlib.Path, monkeypatch: MonkeyPatch) -> N
 @pytest.fixture(autouse=True)
 def configure_ddp(request: pytest.FixtureRequest):
     backend = None
+    use_deepspeed = False
     for item in request.session.items:
-        marker = item.get_closest_marker('gpu')
-        if marker is not None:
+        gpu_marker = item.get_closest_marker('gpu')
+        deepspeed_marker = item.get_closest_marker('deepspeed')
```

Comment:
can you raise an error if a test is marked as both gpu and deepspeed? I think they should be mutually exclusive?